packages/teuchos/comm/test/StackedTimer/stacked_timer.cpp

```diff
@@ -0,0 +1,104 @@
+// @HEADER
+// @HEADER
+
+#include "Teuchos_UnitTestHarness.hpp"
+#include "Teuchos_TimeMonitor.hpp"
+#include "Teuchos_StackedTimer.hpp"
+#include "Teuchos_DefaultMpiComm.hpp"
+#include "Teuchos_GlobalMPISession.hpp"
+#include "Teuchos_DefaultSerialComm.hpp"
+
+#include <thread> // std::this_thread::sleep_for;
+Teuchos::RCP<Teuchos::Comm<int> > comm;
+
+int main(int argc, char **argv) {
+  int result;
+  Teuchos::GlobalMPISession mpiSession(&argc,&argv);
+#ifdef HAVE_MPI
+  comm = Teuchos::rcp(new Teuchos::MpiComm<int>(Teuchos::opaqueWrapper(MPI_COMM_WORLD)));
```

Comment:
You don't need opaqueWrapper any more.  MpiComm has a constructor that takes an `MPI_Comm`.  I wrote it :-) .